pkg/kubelet/kubelet.go

```diff
@@ -1687,6 +1752,17 @@ func (kl *Kubelet) syncPod(o syncPodOptions) error {
 
 	// Call the container runtime's SyncPod callback
 	result := kl.containerRuntime.SyncPod(pod, podStatus, pullSecrets, kl.backOff)
+
+	// TODO(Alban): use helper function and constants
+	userns, _ := pod.Annotations["alpha.kinvolk.io/userns"]
+	klog.V(4).Infof("pod userns setting: %v", userns)
+	if userns == "enabled" {
+		if err := kl.chownDirForRemappedIDs(kl.getPodVolumesDir(pod.UID)); err != nil {
+			kl.recorder.Eventf(pod, v1.EventTypeWarning, events.FailedMountVolume, "Unable to set ownership on mount volumes for pod %q: %v", format.Pod(pod), err)
+			klog.Errorf("Unable to chown volumes for pod %q: %v; skipping pod", format.Pod(pod), err)
+			return err
+		}
+	}
```

Comment:
I think this operation has to be done before L1754, otherwise the container could be started *before* settings the right permissions on the volumes.